app/controllers/tweets_controller.rb

```diff
@@ -0,0 +1,71 @@
+class TweetsController < ApplicationController
+
+  get '/tweets' do
+    if !!session[:id]
+      @user = User.find(session[:id])
+      erb :'tweets/tweets'
+    else
+      redirect to "/login"
+    end
+  end
+
+  get '/tweets/new' do
+    if !session[:id]
+      redirect to "/login"
+    else
+      erb :'tweets/create_tweet'
+    end
+  end
+
+  post '/tweets' do
+    if params.has_value?("")
+      redirect to "/tweets/new"
+    else
+      @user = User.find(session[:id])
+      @tweet = Tweet.create(params)
+      @user.tweets << @tweet
+      redirect to "/tweets/#{@tweet.id}"
+    end
+  end
+
+  get '/tweets/:id' do
+    if !!session[:id]
+      @user = User.find(session[:id])
+      @tweet = Tweet.find_by(params[:id])
+      erb :'tweets/show_tweet'
+    else
+      redirect to "/login"
+    end
+  end
+
+  get '/tweets/:id/edit' do 
+    if !!session[:id]
+      @user = User.find(session[:id])
+      @tweet = Tweet.find_by(params[:id])
```

Comment:
Shouldn't you verify a tweet belongs to the `current_user` (not just that any user is logged in) before allowing them to edit?